include/alpaka/mem/buf/cuda/Copy.hpp

```diff
@@ -407,27 +409,33 @@ namespace alpaka
     #pragma clang diagnostic push
     #pragma clang diagnostic ignored "-Wexit-time-destructors"
 #endif
-                        static std::set<std::pair<int, int>> enabledPeerAccessDevices;
+                        static std::set<std::pair<int, int>> alreadyCheckedPeerAccessDevices;
 #if BOOST_COMP_CLANG
     #pragma clang diagnostic pop
 #endif
                         auto const devicePair = std::make_pair(devSrc, devDst);
 
-                        if(enabledPeerAccessDevices.find(devicePair) == enabledPeerAccessDevices.end())
+                        if(alreadyCheckedPeerAccessDevices.find(devicePair) == alreadyCheckedPeerAccessDevices.end())
                         {
-                            int canAccessPeer=0;
+                            int canAccessPeer = 0;
                             ALPAKA_CUDA_RT_CHECK(cudaDeviceCanAccessPeer(&canAccessPeer, devSrc, devDst));
                             if(!canAccessPeer) {
```

Comment:
maybe the already checked pair, which does not have P2P access, also can be added to the `alreadyCheckedPeerAccessDevices`. 
A `cudaDeviceCanAccessPeer` takes about 20-30µs on a Kepler+Haswell system. On a Pascal/Volta+Broadwell system, this goes down to 1µs ... about the same time what a lookup takes.

Reply:
Yes, this was my intention and therefore the renaming. I simply forgot to add it to the list.

Reply:
fixed